### ISSUE TITLE
Intersesction between Line and Circle

### ISSUE THREAD

**Issue by mjfwest** [scikit-geometry/scikit-geometry#74]
The documentaion at [documentation](https://github.com/scikit-geometry/scikit-geometry/blob/9511b333bbfbb66a47839a5d8aa77729dc3cbe15/docs/source/introduction.ipynb) says :

> It's possible to find intersections between two geometric primitives by using the intersection function

Both the Circle2 and the Line2 are geometric primitives, but trying the intersection between a `Line2` and  a `Circle2` gives the following type error.

```
TypeError: intersection(): incompatible function arguments. The following argument types are supported:
    1. (arg0: skgeom._skgeom.Line2, arg1: skgeom._skgeom.Line2) -> object
    2. (arg0: skgeom._skgeom.Line2, arg1: skgeom._skgeom.Segment2) -> object
    3. (arg0: skgeom._skgeom.Segment2, arg1: skgeom._skgeom.Segment2) -> object
    4. (arg0: skgeom._skgeom.Ray2, arg1: skgeom._skgeom.Segment2) -> object
```

It would appear that it is not possible to find intersections between any two geometric primitives by using the intersection function?

**Comment by wolfv** [comment]
Yeah, we might have to add the overload to the python bindings: https://github.com/scikit-geometry/scikit-geometry/blob/634270b782ed35989741136f203a4aeeedfb1b8c/src/kernel.cpp#L633-L636

Do you want to give that a try?

**Comment by mjfwest** [comment]
I will if I have time. I've solved my imediate problem using sympy.geometry

**Comment by wolfv** [comment]
Yes, that would probably work!

**Comment by richardalligier** [comment]
Hello,

I would like to be able to compute the intersection between Line and Circle. I was under the impression from the messages above, that it was just a one-liner modification in your github code ?!

If this is true, if you have time, it would be really nice to actually do it as computing the intersection between a line and a circle is required in a lot of geometry problems. 

Thank You !
Best,
Richard
